### PR TITLE
Feature/smar-2708 add additional configuration for deployments in helm

### PR DESCRIPTION
**Pull request by egid-fertal-inno**
Requirment was that we should be able to configure via helmchart those two deployment properites:

updateStrategy
revisionHistoryLimit

It was added to values file as:

revisionHistoryLimit:
updateStrategy: {}

when they are empty, default value is used

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@
 - Added test pods supporting `helm test` command
   - There are multiple test pods, some of which run based on what is possible to test in the selected chart configuration
 - Added support for tenant management API (ingress)
+- Make deployment revisionHistoryLimit and updateStrategy configurable via values
```
Comment (FooRider) [inline CHANGELOG.md:10]:
```suggestion
- Deployment revisionHistoryLimit and updateStrategy are configurable via values
```

File: templates/_matcher.tpl [hunk 2]
```diff
@@ -20,6 +20,10 @@ metadata:
     {{- end }}
 spec:
   replicas: {{ .Values.matcher.replicas }}
+  revisionHistoryLimit: {{ .Values.revisionHistoryLimit }}
```
Comment (FooRider) [inline templates/_matcher.tpl:23]: I think we should skip the field entirely in case `.Values.revisionHistoryLimit` is `nill`

File: values.yaml [hunk 3]
```diff
@@ -19,6 +19,17 @@ podLabels: {}
...
+# -- Common revisionHistoryLimit for all deployments
+revisionHistoryLimit:
+# -- Common updateStrategy for all deployments
+## e.g:
+## updateStrategy:
+##  type: RollingUpdate
+##  rollingUpdate:
+##    maxSurge: 25%
+##    maxUnavailable: 25%
+##
```
Comment (FooRider) [inline values.yaml:31]: No need for the example in comment

Comment (egid-fertal-inno) [reply values.yaml]: done :)

**Review by FooRider — APPROVED** [review]

**Review by Temppus — APPROVED** [review]

**Pull request merged**